### DISCUSSION TITLE
Teensy 4.1 & 8bit/16bit displays?

### DISCUSSION THREAD

**Discussion by phillmj** [Bodmer/TFT_eSPI#2696] [Compatible displays and setup files]
Hi, I'm thinking of using a 480x320 3.2" display with a Teensy 4.1, and I notice they all seem to have 16bit interfaces with the 9481 driver.

Would the TFT_eSPI library work with these kinds of displays, such as this one: https://www.aliexpress.com/item/32629717462.html?spm=a2g0o.cart.0.0.7e7238da8ed5oE&mp=1

I've run across some discussion of 8bit/16bit support on your site here, and I think this will not work. But I wanted to ask before wasting a lot of time and money buying hardware that is doomed from the get go.

Thanks,

Mike

**Comment by Bodmer** [comment]
There is no parallel interface support for Teensy boards built into the library. The Teensy 4.1 will work well with SPI interface displays.

Due to lack of familiarity with the Teensy boards and other priorities I don't plan to add parallel support.

Nb. The RP2040 would be able to drive those 16-bit displays and be able to use DMA for screen updates.

**Comment by phillmj** [comment]
I understand. That's what I thought based on your info here, but I wasn't 100% sure and I really like the higher pixel density. I don't know that I can shift to using the RP2040 but I'll look into it. Your library is so important to what I'm doing that I wouldn't change hardware that didn't support it.

A lot of people love the Teensy, so I wonder if at some point you will take a look at them. Not for me, here, but just in general. They are blazing fast, start up super quick, and, well, very tiny or "teensy" lol. And there's a rich community of super smart geniuses supporting it; you'd probably get on well with that lot.

Thanks again for your time and always rushing to support everyone. You're the best, and your lib is amazing.

Mike

**Comment by steven-law** [comment]
I´m absolutly in for teensy support :)

**Reply by Bodmer** [reply]
The Teensy is treated as a generic processor, so is is supported when used with 4 wire SPI displays.

The Teensy board support package seems to handle SPI data efficiently, so performance with the standard Arduino SPI library calls is good..

**Comment by phillmj** [comment]
You are correct; it's a GREAT processor! And fast too!

The lament is I cannot use 8 bit or 16 bit with any screens with the library, even though I know it's possible because some folks on the Teensy forum have made LCD libraries that support them. But I've invested over a year and hundreds/thousands of hours of development with your amazing library, and I wouldn't change it for anything because it's amazing.

Thanks again Mr Bodmer. I feel funny not knowing your first name btw; you've done so much for me and for all of us with your genius libraries.

Best,

Mike

**Reply by Bodmer** [reply]
Post links to the Teensy parallel TFT libraries you found so I can see if there are some efficient low level calls that could be used to implement parallel support. Maybe it would be simple to add. The main problem I have found with very high performance processors is in the management of the strobe timings to avoid exceeding the various timing limits of the display interface.

**Reply by phillmj** [reply]
Here's one: 

https://github.com/mjs513/ILI9488_t3

Here's an excerpt from a forum thread:

"

RA8875 - The library that we are playing with comes from: https://github.com/sumotoy/RA8875
However the owner of this library and many other displays libraries, does not appear to be active anymore... 
He does have a nice WIKI for it: https://github.com/sumotoy/RA8875/wiki

So a few of us have added additional support. The most up to date version is up at: https://github.com/mjs513/RA8875/tree/RA8875_t4

ILI9488 - Yes we do have a version of this that works... More like 3.5" I think? https://github.com/mjs513/ILI9488_t3

We also have a really pain in rear RPI 3.5" dsiplay KeDei, that we spent far more time on than it was worth... 
https://github.com/KurtE/KeDeiRPI35_t3

"

That excerpt came from:

https://forum.pjrc.com/threads/59218-Teensy-4-0-4-3-quot-Display

And some additional discussion around it:

https://forum.pjrc.com/threads/66712-Teensy-4-0-4-1-amp-ILI9488-LCD

It's amazing you would even consider this, even if nothing comes of it or it's too much time -- heartfelt thanks.

**Comment by phillmj** [comment]
Another interesting lib as well:

https://github.com/lvgl/lvgl

Website:

https://lvgl.io/

Quick YouTube demo:

https://www.youtube.com/watch?v=ueWSo2Kv52M

**Comment by phillmj** [comment]
And another lib:

https://github.com/david-res/ILI948x_t41_p

And another:

https://github.com/vmalepati1/Teensy-ILI9341-8-bit-Parallel-Library/blob/master/Adafruit_ILI9341_8bit.h

And a post with data from another guy who's got 16bit working:

https://forum.pjrc.com/threads/47899-Recommending-a-Parallel-Display-for-your-next-project?p=160501&viewfull=1#post160501
